File: jiuzhang/Nine Chapters/2 Binary Search & Sorted Array/java/FindMinimuminRotatedSortedArray.java
public class Solution {
    /**
     * @param num: a rotated sorted array
     * @return: the minimum number in the array
     */
    public int findMin(int[] num) {
        // write your code here
        int l = 0;
        int r = num.length - 1;
        int m;
        while (l <= r) {
            m = (l + r) / 2;
            if (num[m] < num[r]) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return num[r];
    }
}

File: oj/lintcode/153-159-Find-Minimum-in-Rotated-Sorted-Array/FindMinimuminRotatedSortedArray.py
class Solution:
    # @param num: a rotated sorted array
    # @return: the minimum number in the array
    def findMin(self, num):
        # write your code here
        l, r = 0, len(num) - 1
        while l <= r:
            m = (l + r) /2
            if num[m] < num[r]:
                r = m
            else:
                l = m + 1
        return num[r]
